src/main/java/com/conveyal/datatools/manager/jobs/DeployJob.java

```diff
@@ -330,6 +409,242 @@ public void jobFinished () {
+            runInstancesRequest.setIamInstanceProfile(instanceProfile);
+        }
+        final List<Instance> instances = ec2.runInstances(runInstancesRequest).getReservation().getInstances();
+
+        List<String> instanceIds = getIds(instances);
+        Map<String, String> instanceIpAddresses = new HashMap<>();
+        // Wait so that create tags request does not fail because instances not found.
+        try {
+            Waiter<DescribeInstanceStatusRequest> waiter = ec2.waiters().instanceStatusOk();
+//            ec2.waiters().systemStatusOk()
+            long beginWaiting = System.currentTimeMillis();
+            waiter.run(new WaiterParameters<>(new DescribeInstanceStatusRequest().withInstanceIds(instanceIds)));
+            LOG.info("Instance status is OK after {} ms", (System.currentTimeMillis() - beginWaiting));
+        } catch (Exception e) {
+            LOG.error("Waiter for instance status check failed.", e);
+            status.fail("Waiter for instance status check failed.");
+            // FIXME: Terminate instance???
+            return Collections.EMPTY_LIST;
+        }
+        for (Instance instance : instances) {
+            // The public IP addresses will likely be null at this point because they take a few seconds to initialize.
+            instanceIpAddresses.put(instance.getInstanceId(), instance.getPublicIpAddress());
+            String serverName = String.format("%s %s (%s) %d", deployment.r5 ? "r5" : "otp", deployment.name, dateString, serverCounter++);
+            LOG.info("Creating tags for new EC2 instance {}", serverName);
+            ec2.createTags(new CreateTagsRequest()
+                    .withTags(new Tag("Name", serverName))
+                    .withTags(new Tag("projectId", deployment.projectId))
+                    .withResources(instance.getInstanceId())
+            );
+        }
+        List<Instance> updatedInstances = new ArrayList<>();
+        while (instanceIpAddresses.values().contains(null)) {
+            LOG.info("Checking that public IP addresses have initialized for EC2 instances.");
+            // Reset instances list so that updated instances have the latest state information (e.g., public IP has
+            // been assigned).
+            updatedInstances.clear();
+            // Check that all of the instances have public IPs.
+            DescribeInstancesRequest describeInstancesRequest = new DescribeInstancesRequest().withInstanceIds(instanceIds);
+            List<Reservation> reservations = ec2.describeInstances(describeInstancesRequest).getReservations();
+            for (Reservation reservation  : reservations) {
+                for (Instance instance : reservation.getInstances()) {
+                    instanceIpAddresses.put(instance.getInstanceId(), instance.getPublicIpAddress());
+                    updatedInstances.add(instance);
+                }
+            }
+            try {
+                Thread.sleep(10000);
```

Comment:
Would be nice to have a log statement here to indicate that the program is still running.